### PR TITLE
Check cluster expansion status by numsegments

### PR DESCRIPTION
**Pull request by zhangjackey**
If the cluster is in expansion mode, then there must be some partial
tables which are not equal between numsegments and cluster size.
Now we check the expansion status by table's numsegments when we run
gpexpand, gpexpand will raise an error if there are partial tables.

## Here are some reminders before you submit the pull request
- [  ] Add tests for the change
- [  ] Document changes
- [  ] Communicate in the mailing list if needed
- [  ] Pass `make installcheck`

### CODE REVIEW

File: src/test/regress/sql/partial_table.sql [hunk 1]
```diff
@@ -584,3 +584,8 @@ rollback;
 -- to perform distributed commit on the other segments.
 --
 insert into r1 (c4) values (pg_relation_size('r2'));
+
+-- start_ignore
+reset search_path;
+drop schema test_partial_table cascade;
+-- end_ignore
```
Comment (pivotal-ning-yu) [inline src/test/regress/sql/partial_table.sql]: Is this cleanup also need in reshuffle*.sql ?

Comment (zhangjackey) [reply src/test/regress/sql/partial_table.sql]: Thanks for the review.
The tables in reshuffle*.sql are either dropped or expanded, it is no need to clean.

File: contrib/pg_upgrade/check_gp.c [hunk 2]
```diff
@@ -32,12 +33,81 @@ static void check_orphaned_toastrels(void);
...
+	{
+		PGresult   *res;
+		int			ntups;
+		DbInfo	   *active_db = &old_cluster.dbarr.dbs[dbnum];
+		PGconn	   *conn;
+
+		conn = connectToServer(&old_cluster, active_db->db_name);
+		res = executeQueryOrDie(conn,
+								"SELECT true AS expansion "
+								"FROM gp_distribution_policy d "
```
[the remark this answers is not in the document]
Comment (zhangjackey) [reply contrib/pg_upgrade/check_gp.c]: Thanks, I will do it.

File: src/test/regress/sql/partial_table.sql (hunk 1 above)
Comment (ashwinstar) [reply src/test/regress/sql/partial_table.sql]: Add comment why this drop is necessary will help for future, to know the reason.

File: contrib/pg_upgrade/check_gp.c [hunk 3]
```diff
@@ -32,12 +33,81 @@ static void check_orphaned_toastrels(void);
...
+ */
+static void
+check_online_expansion(void)
+{
+	bool		expansion = false;
+	int			dbnum;
+
+	/*
+	* Only need to check cluster expansion status in gpdb6 or later.
+	*/
```
Comment (danielgustafsson) [inline contrib/pg_upgrade/check_gp.c]: Comment indentation

File: contrib/pg_upgrade/check_gp.c [hunk 4]
```diff
@@ -32,12 +33,81 @@ static void check_orphaned_toastrels(void);
...
+	if (GET_MAJOR_VERSION(old_cluster.major_version) < 804)
+		return;
+
+	/*
+	* We only need to check the cluster expansion status on master.
+	* In the other hand the status can not be detected correctly on segments.
```
Comment (danielgustafsson) [inline contrib/pg_upgrade/check_gp.c]: Did you mean "On the other hand.."? Also indentation

Comment (zhangjackey) [reply contrib/pg_upgrade/check_gp.c]: Thanks, I will improve it.

**Pull request merged**